filesystem/src/main/java/org/frankframework/filesystem/LocalFileSystem.java

```diff
@@ -273,8 +280,45 @@ public Date getModificationTime(Path f) throws FileSystemException {
 
 	@Override
 	@Nullable
-	public Map<String, Object> getAdditionalFileProperties(Path f) {
-		return null;
+	public Map<String, Object> getAdditionalFileProperties(Path file) throws FileSystemException {
```

Comment:
Just a question: where exactly does the local filesystem store the extra file properties? FAT/ext2 do not support user defined properties, afaik.

Reply:
It's filesystem-dependent if this is supported or not. But FAT/ext2 are not used by anyone anymore for any serious work, except FAT32 is still used on removable media.
NTFS, ext3, ext4 and many other filesystems support it.

There's a a method to query the FS if it supports the user-defined attributes but I decided not to overcomplicate things.

Reply:
Ah oke, I didn't know. Thanks for the explanation.